File: check-requirements.js
#!/usr/bin/env node

const { execSync } = require('child_process');
const fs = require('fs');

console.log('\n========================================');
console.log('Programming Language Requirements Scanner');
console.log('========================================\n');

const tools = [
  { name: 'Node.js', command: 'node --version' },
  { name: 'npm', command: 'npm --version' },
  { name: 'Python', command: 'python --version' },
  { name: 'pip', command: 'pip --version' },
  { name: 'Java', command: 'java -version' },
  { name: 'C++ (g++)', command: 'g++ --version' },
  { name: 'C (gcc)', command: 'gcc --version' },
  { name: 'Git', command: 'git --version' },
  { name: 'Docker', command: 'docker --version' },
  { name: 'Ruby', command: 'ruby --version' },
  { name: 'Go', command: 'go version' },
  { name: 'Rust', command: 'rustc --version' },
  { name: 'PHP', command: 'php --version' },
  { name: '.NET', command: 'dotnet --version' },
];

let installed = 0;
let notInstalled = 0;

tools.forEach(tool => {
  try {
    const version = execSync(tool.command, { 
      encoding: 'utf-8',
      stdio: ['pipe', 'pipe', 'pipe']
    }).split('\n')[0].trim();
    
    console.log(`✓ ${tool.name.padEnd(20)} : ${version}`);
    installed++;
  } catch (error) {
    console.log(`✗ ${tool.name.padEnd(20)} : Not installed`);
    notInstalled++;
  }
});

console.log('\n========================================');
console.log('SUMMARY');
console.log('========================================');
console.log(`Total Installed: ${installed}`);
console.log(`Total Not Installed: ${notInstalled}`);
console.log(`Scan completed at: ${new Date().toLocaleString()}`);
console.log('========================================\n');

File: check-requirements.py
#!/usr/bin/env python3
"""
Programming Language Requirements Scanner
Checks for installed programming languages and development tools
"""

import subprocess
import sys
from datetime import datetime

def check_tool(name, command, version_flag="--version"):
    """Check if a tool is installed and get its version"""
    try:
        result = subprocess.run(
            f"{command} {version_flag}",
            shell=True,
            capture_output=True,
            text=True,
            timeout=5
        )
        
        if result.returncode == 0:
            version = result.stdout.split('\n')[0].strip() or result.stderr.split('\n')[0].strip()
            return True, version
        else:
            return False, "Not installed"
    except Exception as e:
        return False, "Not installed"

def main():
    print("\n" + "="*40)
    print("Programming Language Requirements Scanner")
    print("="*40 + "\n")
    
    tools = [
        ("Node.js", "node", "--version"),
        ("npm", "npm", "--version"),
        ("Python", "python", "--version"),
        ("pip", "pip", "--version"),
        ("Java", "java", "-version"),
        ("C++ (g++)", "g++", "--version"),
        ("C (gcc)", "gcc", "--version"),
        ("Git", "git", "--version"),
        ("Docker", "docker", "--version"),
        ("Ruby", "ruby", "--version"),
        ("Go", "go", "version"),
        ("Rust", "rustc", "--version"),
        ("PHP", "php", "--version"),
        (".NET", "dotnet", "--version"),
    ]
    
    installed = 0
    not_installed = 0
    
    for name, command, version_flag in tools:
        is_installed, version = check_tool(name, command, version_flag)
        
        if is_installed:
            print(f"✓ {name.ljust(20)} : {version}")
            installed += 1
        else:
            print(f"✗ {name.ljust(20)} : Not installed")
            not_installed += 1
    
    print("\n" + "="*40)
    print("SUMMARY")
    print("="*40)
    print(f"Total Installed: {installed}")
    print(f"Total Not Installed: {not_installed}")
    print(f"Scan completed at: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
    print("="*40 + "\n")

if __name__ == "__main__":
    main()
